CountingBloomFilter.hpp

```diff
@@ -319,25 +308,25 @@ CountingBloomFilter<T>::readHeader(FILE* fp)
 	FileHeader header;
 	if (fread(&header, sizeof(struct FileHeader), 1, fp) != 1) {
 		cerr << "Failed to read header\n";
-		exit(1);
+		exit(EXIT_FAILURE);
 	}
 	if (header.hlen != sizeof(FileHeader)) {
 		cerr << "Bloom Filter header length: " << header.hlen
 		     << " does not match expected length: " << sizeof(FileHeader)
 		     << " (likely version mismatch)" << endl;
-		exit(1);
+		exit(EXIT_FAILURE);
 	}
-	char magic[9];
+	char magic[magicLen];
 	memcpy(magic, header.magic, 8);
 	magic[8] = '\0';
-	if (strcmp(magic, "BLOOMCOU")) {
+	if (strcmp(magic, MAGIC_HEADER_STRING)) {
```

Comment:
```suggestion
	if (strcmp(header.magic, MAGIC_HEADER_STRING) != 0) {
```

Reply:
```suggestion
	if (memcpy(magic, MAGIC_HEADER_STRING, strlen(MAGIC_HEADER_STRING)) != 0) {
```
As discussed in person.